### PR TITLE
Reasoner Explanation BDD Tests

### LINKED ISSUE #95: Implement missing BDD test added for explanation tests
In this PR: https://github.com/graknlabs/verification/pull/33, we added 2 new BDD steps. However, we only implemented them in Grakn Core, and we had to create empty "placeholders" for them in this PR: https://github.com/graknlabs/client-java/pull/94.

We now need to implement these 2 BDD steps in Client Java.

### PR DESCRIPTION
**Pull request by jmsfltchr**
## What is the goal of this PR?

Introduce BDD testing for the explanations returned to the client.

## What are the changes implemented in this PR?

- The major step introduced is `answers contain explanation tree`, which checks the full tree of nested explanations are as expected
- This method is longer and more complicated than we would like, as a result of the nested structure that is being tested
- Note that the major limitation of this testing is that it can only assert one of the possible reasoner resolution paths, and so the test examples need to be carefully picked so that there is only one path, or that the possible paths look identical.

See [https://github.com/graknlabs/verification/pull/33](https://github.com/graknlabs/verification/pull/33)

Fixes #95

### CODE REVIEW

File: test/behaviour/graql/GraqlSteps.java [hunk 1]
```diff
@@ -149,57 +164,222 @@ public void answer_quantity_assertion(int expectedAnswers) {
...
+    @Then("uniquely identify answer concepts")
+    public void uniquely_identify_answer_concepts(List<Map<String, String>> answersIdentifiers) {
+        assertEquals(
+                String.format("The number of identifier entries (rows) should match the number of answers, but found %d identifier entries and %d answers",
+                        answersIdentifiers.size(), answers.size()),
+                answersIdentifiers.size(), answers.size()
+        );
 
         for (ConceptMap answer : answers) {
+            List<Map<String, String>> matchingIdentifiers1 = new ArrayList<>();
```
[the remark this answers is not in the document]
Comment (flyingsilverfin) [reply test/behaviour/graql/GraqlSteps.java:278]: still there!!

**Comment by flyingsilverfin** [comment]
@jmsfltchr  approved with 2 small changes requested (one is for CI that we should do)

File: test/behaviour/graql/GraqlSteps.java [hunk 2]
```diff
@@ -149,57 +164,222 @@ public void answer_quantity_assertion(int expectedAnswers) {
...
+                default:
+                    throw new RuntimeException(String.format("Unrecognised identifier check \"%s\"", check));
+            }
+        }
+    }
 
-    @Then("answer concepts all have key: {word}")
-    public void answer_concepts_have_key_labeled(String label) {
-        answerConceptKey = label;
+    private interface UniquenessCheck {
```
[the remark this answers is not in the document]
Comment (jmsfltchr) [reply test/behaviour/graql/GraqlSteps.java:190]: They do look like they might be generally useful, though I don't know what we'll need them for and hence where they should live

[the remark this answers is not in the document]
Comment (jmsfltchr) [reply test/behaviour/graql/GraqlSteps.java]: For now they're just going out of the way at the bottom of GraqlSteps, we'll move them later if it gets out of hand

**Comment by flyingsilverfin** [comment]
looks good!

**Pull request merged**